source/extensions/tracers/opentelemetry/samplers/dynatrace/sampler_config_provider.cc

```diff
@@ -35,9 +35,12 @@ SamplerConfigProviderImpl::SamplerConfigProviderImpl(
     Server::Configuration::TracerFactoryContext& context,
     const envoy::extensions::tracers::opentelemetry::samplers::v3::DynatraceSamplerConfig& config)
     : cluster_manager_(context.serverFactoryContext().clusterManager()),
-      http_uri_(config.http_uri()),
-      authorization_header_value_(absl::StrCat("Api-Token ", config.token())),
-      sampler_config_(config.root_spans_per_minute()) {
+      http_uri_(config.http_service().http_uri()), sampler_config_(config.root_spans_per_minute()) {
+
```

Comment:
I think we should do some API validation (and deprecated API conversion):

1. if both http_uri and http_service are set, throw.
2. if both http_uri and http_service are not set, throw.
3. if http_uri is set then convert it to http_service.